crates/uv-workspace/src/dependency_groups.rs

```diff
@@ -69,80 +136,123 @@ impl FlatDependencyGroups {
                         }
                     }
                     DependencyGroupSpecifier::IncludeGroup { include_group } => {
-                        resolve_group(resolved, groups, include_group, parents)?;
-                        requirements
-                            .extend(resolved.get(include_group).into_iter().flatten().cloned());
+                        resolve_group(resolved, groups, settings, include_group, parents)?;
+                        if let Some(included) = resolved.get(include_group) {
+                            requirements.extend(included.requirements.iter().cloned());
+                            let versions = full_requires_python
+                                .into_iter()
+                                .flatten()
+                                .chain(included.requires_python.clone().into_iter().flatten())
+                                .collect::<Vec<_>>();
+                            let new_requires_python = VersionSpecifiers::from_iter(versions);
+                            full_requires_python =
+                                (!new_requires_python.is_empty()).then_some(new_requires_python);
+                        }
                     }
                     DependencyGroupSpecifier::Object(map) => {
-                        return Err(DependencyGroupError::DependencyObjectSpecifierNotSupported(
-                            name.clone(),
-                            map.clone(),
-                        ));
+                        return Err(
+                            DependencyGroupErrorInner::DependencyObjectSpecifierNotSupported(
+                                name.clone(),
+                                map.clone(),
+                            ),
+                        );
                     }
                 }
             }
+
+            let empty_settings = DependencyGroupSettings::default();
+            let DependencyGroupSettings { requires_python } =
+                settings.get(name).unwrap_or(&empty_settings);
+            if let Some(requires_python) = requires_python {
+                // Merge in this requires-python to the full flattened version
+                let versions = full_requires_python
+                    .into_iter()
+                    .flatten()
+                    .chain(requires_python.clone())
+                    .collect::<Vec<_>>();
+                let new_requires_python = VersionSpecifiers::from_iter(versions);
```

Comment:
Should there be a method on `VersionSpecifiers` to do this extension? It's a bit clunky doing it here twice?